homeassistant/components/weatherflow_cloud/config_flow.py

```diff
@@ -0,0 +1,108 @@
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+from .const import (
+    _LOGGER,
+    CONF_FIRMWARE_REVISION,
+    CONF_SERIAL_NUMBER,
+    CONF_STATION_ID,
+    DOMAIN,
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WeatherFlowCloud."""
+
+    VERSION = 1
+
+    async def _async_validate_and_get_station_info(
+        self, station_id: int, api_token: str
+    ) -> WeatherFlowStationData:
+        """Validate credentials and obtain station information."""
+
+        session = async_create_clientsession(self.hass)
+        api = WeatherFlow(station_id=station_id, api_token=api_token, session=session)
+        station_info = await api.async_get_station()
+        return station_info
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
+        if user_input is None:
+            return await self._show_setup_form(user_input)
+
+        errors = {}
+
+        try:
+            station_data = await self._async_validate_and_get_station_info(
+                user_input[CONF_STATION_ID], user_input[CONF_API_TOKEN]
+            )
+        except WeatherFlowForecastWongStationId as err:
+            _LOGGER.debug(err)
+            errors["base"] = "wrong_station_id"
+            return await self._show_setup_form(errors)
+        except WeatherFlowForecastBadRequest as err:
+            _LOGGER.debug(err)
+            errors["base"] = "bad_request"
+            return await self._show_setup_form(errors)
+        except WeatherFlowForecastInternalServerError as err:
+            _LOGGER.debug(err)
+            errors["base"] = "server_error"
+            return await self._show_setup_form(errors)
+        except WeatherFlowForecastUnauthorized as err:
+            _LOGGER.debug("401 Error: %s", err)
+            errors["base"] = "wrong_token"
+            return await self._show_setup_form(errors)
+
+        await self.async_set_unique_id(user_input[CONF_STATION_ID])
+        self._abort_if_unique_id_configured()
+
+        return self.async_create_entry(
+            title=station_data.station_name,
+            data={
+                CONF_NAME: station_data.station_name,
+                CONF_STATION_ID: user_input[CONF_STATION_ID],
+                CONF_API_TOKEN: user_input[CONF_API_TOKEN],
+                CONF_DEVICE_ID: station_data.device_id,
+                CONF_FIRMWARE_REVISION: station_data.firmware_revision,
+                CONF_SERIAL_NUMBER: station_data.serial_number,
+            },
+        )
+
+    async def _show_setup_form(self, errors=None):
+        """Show the setup form to the user."""
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_STATION_ID): int, vol.Required(CONF_API_TOKEN): str}
```

Comment:
Can we fetch a list of all stations?

Reply:
I'll have to check the docs ... (Mark as TODO)

Reply:
So it does look like in the docs we have a - get all stations for a user.

<img width="998" alt="image" src="https://github.com/home-assistant/core/assets/6491743/177f4490-d7c7-4ec2-994b-ed3ff2eb9b8b">

So the question is:

- A) User enters credentials - and forecasts get automatically created for all available weather stations... - some what of a redesign - but probably more user friendly at first

- B) User enters credentials - and is then presented a list of possible stations to choose from - they select a station ... if only 1 station its automatically created.

Keeping in mind that the next PR will allow a user to enable Cloud Sensors for a specific weather station - does one option seem better than the other?

If @briis has time to look over this as well - I'd like to get his opinion.

Reply:
I think both would work. I think I would prefer the first. This way is the most simple and I can't really imagine why you don't want to add a certain weather station to your instance
 If you don't want to see it you can disable it.

Reply:
I have only 1 station, so I am not sure if you get an API Token per Station or per user. If it is per station, then option A is the only possibility. If it is per user, I would go for option B, as the users should have the freedom to select what they want to add per station.

Reply:
It's not what they want to add per station, but what station they want to add

Reply:
Based on the API I have to assume that 1 toke gives you multiple stations because of this endpoint:

<img width="270" alt="image" src="https://github.com/home-assistant/core/assets/6491743/cc0f3696-1fbd-421c-a341-b7fb6370fd77">

(Happy Boxing Day if any of you are form England?)